config/routes.rb

```diff
@@ -5,6 +5,7 @@
   resources :notifications, only: %i[create index show]
   resources :status_updates, path: "status-updates", only: %i[create]
   resources :subscriptions, only: %i[create]
+  get "subscribables/:gov_delivery_id", to: "subscribables#show"
```

Comment:
I'm wondering if the `gov_delivery_id` should be a query parameter. Then it makes it easier for us to change this into a more generic search query in the future?

Reply:
I did consider that but decided it was going down the non-standard 'show' path that the `SubscriberListController` already has (the one that I wasted loads of time scratching my head about yesterday that led me to doing this). I think we should stick with the almost restful pass an id (admittedly not 'the' id here) and get the thing. I think the action that we'll eventually add to replace the stuff on `SubscriberListController` would accept parameters and be more like a search type thing. We might have some trouble with versioning but having this separate controller altogether will hopefully help with that. All that said, happy to change if I'm in the minority though.

Reply:
My main concern with that was when we eventually get rid of GovDelivery IDs, it could cause us problems or make it a lot harder to move away from them as what seems like the standard `GET /subscribables/:id` endpoint isn't really using an ID.

Reply:
What do we ultimately want the interface to look like? (i.e. once we get rid of GovDelivery IDs, how will we want this to work)? Probably a silly question, but I'm not really picturing it in my head......

Reply:
I still think we'll need some way of getting the subscribable with 'that' id though. We'll call it something else perhaps but I still don't imagine well just use the incrementing integer id. Like with do already with `Subscription#uuid` for unsubscribing. If we use `subscribables?gov_delivery_id=<id>` then you'd expect that to return a collection as it would be the index by convention. We could return a collection with one thing in and deal with that at the other end but that seems wrong too.

Reply:
I suppose it's analogous to getting a content item by slug. The slug isn't the db id but it's the id that we choose to expose.

Reply:
We've (@thomasleese @bevanloon and me) discussed this offline. It's not ideal but as far as we can see we'll probably end up renaming `gov_delivery_id` but continuing to use a non-DB identifier for this functionality. In that eventuality this won't be too difficult to change and the URL pattern will stay the same, only the parameter name will change internally. (It's already referred to as `topic_id` on the frontend anyway so we've got renaming to do there too in any case). So... we're going to press ahead with this for now.